soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,201 @@
+import main
+import robocup
+import behavior
+import constants
+import enum
+import standard_play
+import evaluation
+import tactics.coordinated_pass
+import skills.move
+import skills.capture
+
+
+class distraction(standard_play.StandardPlay):
+
+    class State (enum.Enum):
+        setup = 1, 'capture ball and move distractor and striker into position'
+        optional_adjustment = 2, 'capture and setup a pass to the center right'
+        center_pass = 3, 'pass to the center right of the field'
+        passing = 4, 'pass from first capture to distractor'
+        cross = 5, 'pass from distractor to striker'
+        shoot = 6, 'shoot at goal'
+
+    def __init__(self):
+        super().__init__(continuous=False)  
+
+        for s in distraction.State:
+            self.add_state(s, behavior.Behavior.State.running)
+
+        self.ball_is_far = main.ball().pos.y < (0.4*constants.Field.Length)
+
+        self.distraction_point = robocup.Point(0.40*constants.Field.Width, 
+            0.95*constants.Field.Length) #the first distraction point
+
+        self.distraction_recieve_pass_point = robocup.Point(0.40*constants.Field.Width, 
+            0.8*constants.Field.Length) #the second distraction point
+
+        self.striker_point = robocup.Point(-0.40*constants.Field.Width, 
+            0.9*constants.Field.Length) #striker's position
+
+        self.center = robocup.Point(0.5*constants.Field.Width,
+            0.5*constants.Field.Length) #center of field position, used if ball is far
+
+        self.distract_box = robocup.Rect(
+            robocup.Point(0.33*constants.Field.Width/2,
+                         (7/9)*constants.Field.Length),
+            robocup.Point(constants.Field.Width, 
+                         constants.Field.Length))
+
+        self.striker_box = robocup.Rect(
+            robocup.Point(-0.33*constants.Field.Width/2,
+                         (7/9)*constants.Field.Length),
+            robocup.Point(-constants.Field.Width, 
+                         constants.Field.Length))
+
+        self.distracter_get_close_ball = self.distract_box.contains_point(main.ball().pos)
+        self.striker_get_close_ball = self.striker_box.contains_point(main.ball().pos)
+
+        self.pass_striker_instead = False
+        self.dont_shoot = False
+
+        self.add_transition(behavior.Behavior.State.start,
+                        distraction.State.setup, 
+                        lambda: not self.ball_is_far,
+                        'if ball is not far')
```

Comment:
These phrases at the end usually are pretty short. They are like 1 to 2 words saying `passing`, `ball captured`, `setting up` etc. Don't delete the stuff though, they are super good comments. Just throw them in front of the `add_transition`.

eg:
```
# comment that's current at the end
self.add_transition(asdf,
                    asdf,
                    lambda: asdf,
                    'short phrase')
```